x-pack/plugins/apm/server/routes/settings/agent_configuration/convert_settings_to_string.ts

```diff
@@ -13,17 +13,27 @@ import { AgentConfiguration } from '../../../../common/agent_configuration/confi
 export function convertConfigSettingsToString(
   hit: SearchHit<AgentConfiguration>
 ) {
-  const config = hit._source;
+  const { settings } = hit._source;
 
-  if (config.settings?.transaction_sample_rate) {
-    config.settings.transaction_sample_rate =
-      config.settings.transaction_sample_rate.toString();
-  }
+  const convertedConfigSettings = {
+    ...settings,
+    ...(settings?.transaction_sample_rate
+      ? {
+          transaction_sample_rate: settings.transaction_sample_rate.toString(),
+        }
+      : {}),
+    ...(settings?.transaction_max_spans
+      ? {
+          transaction_max_spans: settings.transaction_max_spans.toString(),
+        }
+      : {}),
+  };
 
-  if (config.settings?.transaction_max_spans) {
-    config.settings.transaction_max_spans =
-      config.settings.transaction_max_spans.toString();
-  }
-
-  return hit;
+  return {
+    ...hit,
+    _source: {
```

Comment:
I'm not sure if it matters much, but it does feel kind of gross that we're transforming `_source` here. Is not returning _source, but just the data in _source an option, or is that a breaking change?

Reply:
I feel the same, looked to simplify it at first but it's used in the [public API](https://www.elastic.co/guide/en/kibana/master/agent-config-api.html#apm-search-config).